File: fapi/src/main/java/com/artsykov/fapi/models/LogInParam.java
package com.artsykov.fapi.models;

import lombok.Data;
import org.springframework.stereotype.Component;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Pattern;

@Data
public class LogInParam {

    @NotEmpty(message = "Email is required")
    @Email(message = "Email or password is incorrect")
    private String email;

    @NotEmpty(message = "Email or password is required")
    @Pattern(regexp="^[A-Za-z0-9]{8,}$",
            message="Password is invalid")
    private String password;
}

File: frontend/src/app/modules/models/logInParam.ts
export class logInParam {
  email: string;
  password: string;

  public constructor(email: string, password: string) {
    this.email = email;
    this.password = password;
  }

  static cloneBase(signinParam: logInParam): logInParam {
    const clonedSigninParam: logInParam = new logInParam(signinParam.email, signinParam.password);
    return clonedSigninParam;
  }
}
